packages/platform-browser/src/browser/server-transition.ts

```diff
@@ -21,10 +21,11 @@ export function appInitializerFactory(transitionId: string, document: any, injec
     // the server.
     injector.get(ApplicationInitStatus).donePromise.then(() => {
       const dom = getDOM();
-      const styles: any[] =
-          Array.prototype.slice.apply(document.querySelectorAll(`style[ng-transition]`));
-      styles.filter(el => el.getAttribute('ng-transition') === transitionId)
-          .forEach(el => dom.remove(el));
+      const styles: HTMLCollectionOf<HTMLStyleElement> =
+          document.querySelectorAll(`style[ng-transition="${transitionId}"]`);
+      for (let i = 0; i < styles.length; i++) {
+        dom.remove(styles[i]);
```

Comment:
Shouldn't this be?
```suggestion
        dom.remove(styles[0]);
```
(because the loop is moving forward instead of backwards)

Reply:
The list here doesn't update so what I have above is correct (see https://codepen.io/crisbeto/pen/VwbNzNq?editors=1011). What you're describing is only necessary when reading `element.children` or `element.childNodes` as elements are being removed.